### PR TITLE
Created the PID (Proportional-Integrative-Derivative) controller plugin

### PR DESCRIPTION
**Pull request by armstrongmsg**
The PID Controller changes the amount of allocated resources based on the current error value, the difference between the current value and the last received, the accumulated error and three parameters: the proportional, derivative and integrative factors. The actuation size is proportional to the error value and the factors received. The amount of resources to add or remove is calculated as shown in the expressions below:

c<sub>p</sub> = -f<sub>p</sub>*error
c<sub>d</sub> = -f<sub>d</sub>*(error<sub>n</sub> - error<sub>n - 1</sub>)
c<sub>i</sub> = -f<sub>i</sub>*(error<sub>0</sub> + error<sub>1</sub> + ... + error<sub>n - 1</sub> + <sub>n</sub>)
size = c<sub>p</sub>+c<sub>d</sub>+c<sub>i</sub>

Where, c<sub>p</sub> is the proportional component, c<sub>d</sub> is the derivative component, c<sub>i</sub> is the integrative component and size is the amount of resources.  f<sub>p</sub> is the proportional factor parameter, f<sub>d</sub> is the derivative factor parameter and f<sub>i</sub> is the integrative factor.

### CODE REVIEW

File: bigsea-controller/service/api/controller/controller_builder.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 from service.api.controller.plugins.proportional_derivative.proportional_derivative_controller import (
```
Comment (betobetodev) [inline bigsea-controller/service/api/controller/controller_builder.py]: It looks better if you change this line to:
**from service.api.controller.plugins.proportional_derivative.controller import (**

File: bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import datetime
+import time
+
+class PIDAlarm:
+
+    ERROR_METRIC_NAME = "application-progress.error"
+    
+    def __init__(self, actuator, metric_source, trigger_down, trigger_up, min_cap, max_cap, 
+                            metric_rounding, heuristic_options, application_id, instances):
+        # TODO: Check parameters
```
Comment (betobetodev) [inline bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py]: Still **TODO**?

File: bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        self.cap = -1
+    
+    def check_application_state(self):
+        """
+            Checks the application progress by getting progress metrics from a 
+            metric source, checks if the metrics are new and tries to modify the
+            amount of allocated resources if necessary.
+        """
+        
+        # TODO: Check parameters
```
Comment (betobetodev) [inline bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py]: Still **TODO**?

**Review by betobetodev — CHANGES_REQUESTED** [review]
Before any pull request, please check the pep8 errors. This will avoid more requests for change.

File: bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py (hunk 2 above)
Comment (armstrongmsg) [reply bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py]: Yes, I do not think it is going to be solved anytime soon (actually all plugins need this).

File: bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py (hunk 3 above)
Comment (armstrongmsg) [reply bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py]: Yes, I do not think it is going to be solved anytime soon.

File: bigsea-controller/service/api/controller/controller_builder.py (hunk 1 above)
Comment (armstrongmsg) [reply bigsea-controller/service/api/controller/controller_builder.py]: I would prefer to keep the classes names as they are now.

Comment (armstrongmsg) [reply bigsea-controller/service/api/controller/controller_builder.py]: OK, but I will keep the classes names as they are now.

**Review by igornatanael — APPROVED** [review]
I looks good for me. If @robertonscjr has nothing more to add in his review, we can merge it

**Pull request merged**